doc/src/devdocs/ub.md

```diff
@@ -0,0 +1,52 @@
+
+## List of sources of undefined behavior
+
+The following is a list of sources of undefined behavior,
+though it should currently not be considered exhaustive:
+
+- Replacement of `const` values. Note that in interactive mode the compiler will issue a warning for this and some care is taken to mimize impact as a user convenience, but the behavior is not defined.
+- Various modification of global state during precompile. Where possible, this is detected and yields an error, but detection is incomplete.
+- Incorrect implementation of a `Core.OptimizedGenerics` interface [1]
+- Any invocation of undefined behavior in FFI code (e.g. `ccall`, `llvmcall`) according to the semantics of the respective language
+- Incorrect signature types in `ccall` or `cfunction`, even if those signatures happen to yield the correct ABI on a particular platform
+- Incorrect use of annotations like `@inbounds`, `@assume_effects` in violation of their requirements [1]
+- Retention of pointers to GC-tracked objects outside of a `@GC.preserve` region
+- Memory modification of GC-tracked objects without appropriate write barriers from outside of julia (e.g. in native calls, debuggers, etc.)
+- Violations of the memory model using `unsafe` operations (e.g. `unsafe_load` of an invalid pointer, pointer provenance violations, etc)
+- Violations of TBAA guarantees (e.g. using `unsafe_wrap`)
+- Mutation of data promised to be immutable (e.g. in `Base.StringVector`)
+- Data races
+- Modification of julia-internal mutable state (e.g. task schedulers, data types, etc.)
+- A value other than `false` (`reinterpret(UInt8, b) == 0x00`) or `true` (`reinterpret(UInt8, b) == 0x01`) for a `Bool` `b`.
+- Invoking undefined behavior via compiler intrinsics.
+
+[1] Incorrect use here may be UB, even if not semantically executed, please see the specific documentation of the feature.
+
+## Implementation-defined behavior
+Some behavior is technically forbidden by the semantics of the language, but required in certain parts of the implementation and thus allowed as long as implementation-defined constraints are obeyed. Nevertheless, these constructs should be avoided when possible, as the implementation-defined constraints may not be stable across julia versions.
+
+- Construction of objects using `eval(Expr(:new))` rather than their constructors
+
+## Special cases explicitly NOT undefined behavior
+
+- As of Julia 1.11, access to undefined bits types is no longer undefined behavior. It is still allowed to return an arbitrary value of the bits type, but the value returned must be the same for every access and use thereof is not undefined behavior. In LLVM terminology, the value is `freeze undef`, not `undef`.
```

Comment:
Do you remember when this change was implemented? It might mean we are now able to remove the check at https://github.com/JuliaLang/julia/blob/d8b9810b8908560cc4aaa744c0e587f4dfd11bfb/base/compiler/tfuncs.jl#L2493

Reply:
Yes, you're right, that should be removable. We already taint the consistency when allocating the object, so there is no longer any UB with accessing an object (since #52169 in November):
```
julia> struct A; b::Int; A() = new(); end
julia> Base.infer_effects(()) do; A(); end
(!c,+e,+n,+t,+s,+m,+u)
```

Reply:
Wait, if it's assured that an undefined `isbitstype`-field consistently returns the same value, shouldn't those allocations also be considered `:consistent`, unless they are mutable?

Reply:
A specific, known allocation always returns the same value when observed, but a specific call / allocation site does not